### DISCUSSION TITLE
Document manual mesh leveling better

### DISCUSSION THREAD

**Discussion by dafi87** [Jyers/Marlin#83] [Ideas]
Hey there,
I had to play around with configuration options quite a bit, to get manual mesh leveling running.
So I just want to share this here for others who were following some tutorials on it and don't get it to work with this firmware (that is really awesome btw!)

Instead of define MESH_BED_LEVELING, you need to define AUTO_BED_LEVELING_BILINEAR for the manual(!) mesh to work.

This should either be documented or fixed/changed.

**Comment by Jyers** [comment]
I'm planning on updating the configuration page of the wiki to better go over some of the options. I had figured most people would start with the ManualMesh template, hence why there wasn't a ton of info on the setting required. If you're interested why it uses bilinear, its because bilinear provides a marginally more robust system for creating the mesh as well as simplifying some of the code.

**Reply by dafi87** [reply]
Sounds great! I thought there was a reason for you to set it up like this. And I think you are right, the average person would start from the template. I had to dig into the code to look in what cases you enable the level main menu tile instead of the info tile, since it still showed the info tile for me. From there it was easy, haha.
Great work you do!

**Comment by DeNohvo** [comment]
I used the 3x3 mesh firmware and its wonderful, need to fine tune it a bit more but overall it works great. I updated my slicer to add M420 S1

Great work.

**Comment by kjeldahl** [comment]
@Jyers I have a question about how to see that the mesh is loaded/used.
After having saved a mesh, will the values be reused when going into the Leveling menu to update the mesh? Or will it always use 0.20 when entering the menu?
While printing: is the Z value at the bottom supposed to be updated by changes caused by the mesh? I suppose not.

I have been looking at Z axis rod but I cannot see or feel it move though the mesh values varies across the bed. 
I have added M420 S1 to my startup code just after the G28.

**Reply by kjeldahl** [reply]
I have looked in the code and it always start at 0.20 (MANUAL_PROBE_START_Z)

**Reply by Jyers** [reply]
Yes, the leveling process always starts at 0.2 and the screen will not show a change in the z value as it treats the mesh like an offset. I would recommend trying a test print which uses most of the bed and watching the z axis coupler, this is the easiest way to make sure the mesh is being applied.
